loopchain/peer/block_manager.py

```diff
@@ -804,13 +804,8 @@ def add_complain(self, vote: LeaderVote):
                 return
 
             elected_leader = self.epoch.complain_result()
-            if elected_leader:
-                if elected_leader == ExternalAddress.empty().hex_xx() and vote.round_ == self.epoch.round:
-                    util.logger.warning(f"Fail to elect the next leader on {self.epoch.round} round.")
-                    elected_leader = self.blockchain.get_next_rep_in_reps(
-                        ExternalAddress.fromhex(self.epoch.leader_id), self.epoch.reps).hex_hx()
-                if self.epoch.round == vote.round_:
-                    self.__channel_service.reset_leader(elected_leader, complained=True)
+            if elected_leader and self.epoch.round == vote.round_:
```

Comment:
self.epoch.round == vote.round_ this condition already checked in prev line #795, me think.

Reply:
How about this?

```
     def add_complain(self, vote: LeaderVote):
        util.logger.spam(f"add_complain vote({vote})")

        if not self.epoch:
            util.logger.debug(f"Epoch is not initialized.")
            return

        if self.epoch.height == vote.block_height:
            if self.epoch.round == vote.round_:
                self.epoch.add_complain(vote)
                elected_leader = self.epoch.complain_result()
                if elected_leader:
                    self.__channel_service.reset_leader(elected_leader, complained=True)
            elif self.epoch.round > vote.round_ and vote.new_leader != ExternalAddress.empty():
                self.__send_fail_leader_vote(vote)
            else:
                # TODO: do round sync
                pass

        elif self.epoch.height < vote.block_height:
            self.__channel_service.state_machine.block_sync()
```